i18n(it): tutorial - unit 1 setup - part 1 (#4573)
withastro/docs: src/content/docs/it/tutorial/1-setup/1.mdx

---
type: tutorial
title: Prepara il tuo ambiente di sviluppo
description: |-
  Tutorial: Costruisci il tuo primo blog con Astro —
  Installa sul tuo computer gli strumenti di cui avrai bisogno per completare il tutorial
---
import Badge from '~/components/Badge.astro';
import Checklist from '~/components/Checklist.astro';
import Box from '~/components/tutorial/Box.astro';
import MultipleChoice from '~/components/tutorial/MultipleChoice.astro';
import Option from '~/components/tutorial/Option.astro';
import PreCheck from '~/components/tutorial/PreCheck.astro';

<PreCheck>
  - Installa tutti gli strumenti di cui avrai bisogno per costruire il tuo sito Astro
</PreCheck>

## Come ottenere gli strumenti di sviluppo necessari

### Terminale

Userai un **terminale a linea di comando** per creare il tuo progetto Astro e per eseguire i comandi più importanti per assemblare, sviluppare e testare il tuo sito.

> Potrai accedere alla linea di comando usando un programma cosiddetto "Terminale" per il tuo sistema operativo. I terminali più diffusi sono **Terminale** (MacOS/Linux), **Prompt dei comandi** (Windows) o **Termux** (Android). Probabilmente uno di questi è già presente sulla tua macchina.

Reviewer: Non sono sicuro la scriverei in questo modo, personalmente. Probabilmente eviterei il "cosiddetto".

```suggestion
Potrai accedere alla linea di comando usando un programma terminale locale per il tuo sistema operativo. I terminali più diffusi sono **Terminale** (MacOS/Linux), **Prompt dei comandi** (Windows) o **Termux** (Android). Probabilmente uno di questi è già presente sulla tua macchina.
```

### Node.js

> Per eseguire Astro sul tuo computer devi avere installato [**Node.js**](https://nodejs.org/en/), versione `v16.12.0` o successive.

Reviewer: ```suggestion
Per eseguire Astro sul tuo computer devi avere installato [**Node.js**](https://nodejs.org/it/), versione `v18.14.1` o successive.
```

Per verificare se è già presente una versione compatibile sul tuo sistema, esegui il comando seguente nel terminale:

```sh
node -v

// Output di esempio

> v16.14.0

Reviewer: ```suggestion
v18.14.1
```

```

> Se il comando restituisce la versione `v16.12.0` o superiore, tutto a posto, puoi continuare.

Reviewer: ```suggestion
Se il comando restituisce la versione `v18.14.1` o superiore, tutto a posto, puoi continuare.
```

> Se invece il comando restituisce un errore simile a `Command not found` o `Comando non trovato`, oppure una versione inferiore a `v16.12.0`, allora dovrai [installare una versione di Node.js compatibile](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm)

Reviewer: A parte la versione, non so se tenere anche l'inglese o meno. @delucis, cosa ne dici?

```suggestion
Se invece il comando restituisce un errore simile a `Command 'node' not found` o `Comando 'node' non trovato`, oppure una versione inferiore a `v18.14.1`, allora dovrai [installare una versione di Node.js compatibile](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm)
```

:::

1. [Scarica e installa VS Code](https://code.visualstudio.com/#alt-downloads) o qualsiasi altro editor di codice di tuo gradimento. 

<Box icon="question-mark">

### Verifica le tue conoscenze

Quale dei seguenti è...
1. Un editor del codice, per modificare i tuoi file e il loro contenuto?

    <MultipleChoice>
      <Option>
        Browser
      </Option>
      <Option>
        Terminale
      </Option>
      <Option isCorrect>
        VS Code
      </Option>
    </MultipleChoice>

2. Uno strumento di controllo delle versioni online per il tuo codice?

    <MultipleChoice>
      <Option isCorrect>
        GitHub
      </Option>
      <Option>
        Terminale
      </Option>
      <Option>
        VS Code
      </Option>
    </MultipleChoice>

3. Un'applicazione per eseguire comandi?
    <MultipleChoice>
      <Option>
        GitHub
      </Option>
      <Option isCorrect>
        Terminale
      </Option>
      <Option>
        Browser
      </Option>
    </MultipleChoice>

</Box>

<Box icon="check-list">
## Checklist per continuare con i passi successivi

<Checklist>

> - [ ] Ho accesso alla linea di comando in un terminale

Reviewer: Puntiglioso... ma l'ho notato ahah

```suggestion
- [ ] Ho accesso alla linea di comando in un terminale.
```
